external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaTupleListener.java

```diff
@@ -0,0 +1,48 @@
+package org.apache.storm.kafka.spout;
+
+import java.io.Serializable;
+import java.util.List;
+
+
+/**
+ * The KafkaTupleListener handles state changes of a kafka tuple inside a KafkaSpout.
+ */
+public interface KafkaTupleListener extends Serializable {
+
+    /**
+     * Called when the tuple is emitted.
+     *
+     * @param tuple the storm tuple.
+     * @param msgId The id of the tuple in the spout.
+     */
+    void onEmit(List<Object> tuple, KafkaSpoutMessageId msgId);
+
+
+    /**
+     * Called when a tuple is acked.
+     *
+     * @param msgId The id of the tuple in the spout.
+     */
+    void onAck(KafkaSpoutMessageId msgId);
+
+    /**
+     * Called when a fail reaches the spout, but the Kafka record does not belong to the spout anymore.
```

Comment:
I changed the name of the method and updated the comment to reflect your remarks. You were right about the onPartitionsReassigned. That was what I intended with the call.